### DISCUSSION TITLE
How can use a blob/stream as analyzer input?

### DISCUSSION THREAD

**Discussion by jgabriel98** [hvianna/audioMotion-analyzer#83] [Q&A]
So, the thing is: i request my server for a track and it answers with a .mp3 file.

But i cannot plug the `blob.stream()` into _audioMotion-analyzer_:
![image](https://github.com/user-attachments/assets/a7814bc3-9fb3-42b1-89c2-2fcb8abdda7b)

more specifically, i cannot use a `ReadableStream` as a `MediaStream`:
![image](https://github.com/user-attachments/assets/acc0713a-61c4-4b72-bc77-2aa37e7106d0)

So my question is:
**How can i turn a blob/stream into a MediaStream or AudioSouceNode, so i can plug it into audioMotion-analyzer?**

----------------

##### note 2: i only want to visually analyze it, no sound output is required.

**Comment by jgabriel98** [comment]
Or maybe is there a way to analyse a stream directly instead of a AudioNode?

**Reply by hvianna** [reply]
At the moment an AudioNode is required for connection with the AnalyserNode.

**Comment by hvianna** [comment]
The only way I know how to do this is by using an HTMLAudioElement:

```js
const audio = new Audio();
audio.src = URL.createObjectURL( blob );
audio.play();
```

> note: using `<audio />` is not an option, since it requires user input to start playing. And i want to automatically analyze the audio without any user interaction.

If you mean the user gesture required to unlock the AudioContext, any click in the page will do (after audioMotion is instantiated). You can also whitelist your server in the browser's *site settings* to always allow sound playback.

**Reply by jgabriel98** [reply]
> The only way I know how to do this is by using an HTMLAudioElement:
> 
> ```js
> const audio = new Audio();
> audio.src = URL.createObjectURL( blob );
> audio.play();
> ```

Yeah, I'm actually doing like this now. But I really wanted to just automatically analyze the audio.

> If you mean the user gesture required to unlock the AudioContext, any click in the page will do (after audioMotion is instantiated). You can also whitelist your server in the browser's _site settings_ to always allow sound playback.

Got it.
Since my goal is to run it on a android webview,i guess the embedded app can change those settings.

Thanks anyway!
